namesrv/src/main/java/org/apache/rocketmq/namesrv/routeinfo/RouteInfoManager.java

```diff
@@ -678,10 +678,7 @@ public TopicRouteData pickupTopicRouteData(final String topic) {
                     if (null == brokerData) {
                         continue;
                     }
-                    BrokerData brokerDataClone = new BrokerData(brokerData.getCluster(),
-                        brokerData.getBrokerName(),
-                        (HashMap<Long, String>) brokerData.getBrokerAddrs().clone(),
-                        brokerData.isEnableActingMaster(), brokerData.getZoneName());
+                    BrokerData brokerDataClone = new BrokerData(brokerData);
```

Comment:
why not make a deep copy of  a BrokerData

Reply:
> why not make a deep copy of a BrokerData

I see that the constructor does the job @fuyou001